curvefs/src/metaserver/inode_manager.cpp

```diff
@@ -316,22 +309,84 @@ MetaStatusCode InodeManager::GetOrModifyS3ChunkInfo(
             }
         }
         if (!duplicated) {
-            for (auto &item : s3ChunkInfoAdd) {
-                auto it = old.mutable_s3chunkinfomap()->find(item.first);
-                if (it != old.mutable_s3chunkinfomap()->end()) {
-                    MergeToS3ChunkInfoList(item.second, &(it->second));
-                } else {
-                    old.mutable_s3chunkinfomap()->insert(
-                        {item.first, item.second});
+            if (fromS3Compaction) {
+                VLOG(9) << "s3compact: request from s3compaction, inodeId:"
+                        << inodeId;
+                auto requestCheck = [&](bool cond) {
+                    if (!cond)
+                        LOG(ERROR)
+                            << "s3compact: bad GetOrModifyS3ChunkInfo request";
+                    return cond;
+                };
+                if (!requestCheck(s3ChunkInfoAdd.size() ==
+                                  s3ChunkInfoRemove.size()))
+                    return MetaStatusCode::PARAM_ERROR;
+                for (const auto& item : s3ChunkInfoAdd) {
+                    const auto& chunkIndex = item.first;
+                    const auto& toAddList = item.second;
+                    if (!requestCheck(toAddList.s3chunks_size() == 1))
```

Comment:
why s3chunks_size() == 1 is not allowed?

Reply:
only allowed s3chunks_size() == 1
s3compact will compact several s3chunkinfo of a chunkindex into only one.